Generate errors for getters of untyped variables returning nothing (superseded by #79363)

garychia opened the pull request:
<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->
Fixes #63393
Errors are generated for getter functions where not all code paths return a value.
![image](https://user-images.githubusercontent.com/88014292/181275114-ba2ddb42-4654-4fda-a8ec-d58ec1bfb2b3.png)

angad-k committed:
Check parameter validity in `Object::set_script`

Fixes #46120.

adamscott commented:
As the PR needs a rebase and the release is close, I moved the milestone to 4.2.
I put it, though, in the list of PRs that needs a review from the team, not to forget it.

van800 committed:
Update the RiderPathLocator to support the JetBrains Toolbox 2.0

ajreckof committed:
Fix crash when removing child during focus exit.

RandomShaper committed:
Disable doc cache temporarily for pre-RC stabilization

akien-mga committed:
Merge pull request #77532 from dsnopek/text_server_adv_gdextension

Fix text_server_adv compiling as a GDExtension

akien-mga committed:
Merge pull request #78614 from RandomShaper/disable_doc_cache

Disable doc cache temporarily for pre-RC stabilization

YuriSizov committed:
Merge pull request #78611 from YuriSizov/editor-cure-for-early-onset-layout-dementia

Handle contextual editors gracefully when restoring layout

akien-mga committed:
Revert "Fix that the focus-out notification got sent deferred"

This reverts commit 8cdab04d7fd57aaabd790349cd8a4e9ec21a7edd.

Fixes #78589.

clayjohn committed:
Reset filter/repeat state of textures in GL Compatibility renderer when render target is cleared

puchik committed:
Clear specular buffer if bg mode is canvas and ss effects are used

Explicitly clear the separate specular buffer when the background mode is canvas and screen space effects (and thus a separate specular buffer) are used.

Sauermann committed:
Fix documentation of `popup_centered_ratio`

akien-mga committed:
Merge pull request #78591 from ajreckof/Fix-crash-when-removing-child-during-focus-exit-

Fix crash when removing child during focus exit.

akien-mga committed:
Merge pull request #78620 from clayjohn/GL-viewport

Reset filter/repeat state of textures in GL Compatibility renderer when render target is cleared

Calinou committed:
Display a message about missing C# support on Android/iOS/Web platforms

When trying to export a C# project, this displays an error message after
creating a export preset for an unsupported platform.

Support for these platforms is planned for a future release.

Calinou committed:
Use bullet points in shader editor creation dialog

This is consistent with the script creation dialog.

fire committed:
Varying mismatch causing shaders to fail.

akien-mga committed:
Merge pull request #78629 from Calinou/editor-export-mobile-web-csharp-error

Display a message about missing C# support on Android/iOS/Web platforms

akien-mga committed:
Revert "Fix paste value not updated in dictionaries/arrays"

akien-mga committed:
Merge pull request #78643 from godotengine/revert-76711-fix-paste-value-not-updating-element-in-dictionaries/array

Revert "Fix paste value not updated in dictionaries/arrays"

dalexeev committed:
Unexpose `DocCache` class

Sauermann committed:
Fix scene load crash related to `_ready`

The iterator might get invalidated during `_ready`-user-code.
Prevent this by disallowing set_name during iterator-operations.

akien-mga committed:
Merge pull request #78637 from V-Sekai/varying-offset-mismatch

Varying mismatch causing shaders to fail.

smix8 committed:
Fix NavObjects map assignments

Fixes NavObject map assignments.

AThousandShips committed:
Add example for `NodePath` to grandparent

AThousandShips committed:
Clarify restrictions on nested `MultiplayerAPI` in `SceneTree`

KoBeWi committed:
Improve script icon cache

Geometror committed:
[GraphEdit] Fix port hotzones at zoom levels other than 100%

smix8 committed:
Fix GlobalScope doc link for NavigationServer3D

Fixes GlobalScope doc link for NavigationServer3D.

smix8 committed:
Add missing default Callables for new NavigationServer geometry parse and bake functions

Adds missing default Callables for new NavigationServer geometry parse and bake functions.

Commit:
Fixed grid disappearance

AThousandShips committed:
Fix documentation for `Mutex.try_lock`

Documentation was not updated when return type was changed from `Error`
to `bool`

Calinou committed:
Mention Xbox menu button by name in Start button description

Microsoft officially calls it the Menu button:

https://support.xbox.com/en-US/help/hardware-network/controller/get-to-know-your-xbox-series-x-s-controller

Faless committed:
Merge pull request #78662 from AThousandShips/mp_doc

Clarify restrictions on nested `MultiplayerAPI` in `SceneTree`

akien-mga committed:
Merge pull request #78626 from Sauermann/fix-centered-doc

Fix documentation of `popup_centered_ratio`

akien-mga committed:
Merge pull request #78654 from Sauermann/fix-scene-load-crash

Fix scene load crash related to `_ready`

akien-mga committed:
Merge pull request #78660 from AThousandShips/doc_node_path

Add example for `NodePath` to grandparent

akien-mga committed:
Merge pull request #78665 from smix8/fix_navobjects_map_4.x

Fix NavObjects map assignments

akien-mga committed:
Merge pull request #78673 from Geometror/fix-graphnode-hotzones

[GraphEdit] Fix port hotzones at zoom levels other than 100%

akien-mga committed:
Merge pull request #78676 from smix8/doc_globalscope_navserver3d_4.x

Fix GlobalScope doc link for NavigationServer3D

akien-mga committed:
Merge pull request #78677 from smix8/navserver_default_values_4.x

Add missing default Callables for new NavigationServer geometry parse and bake functions

akien-mga committed:
Merge pull request #78700 from AThousandShips/mutex_doc

Fix documentation for `Mutex.try_lock`

Faless committed:
[MP] Fix delta variables index decoding.

akien-mga committed:
Update AUTHORS and DONORS list

New contributors added to AUTHORS:
@ajreckof, @AThousandShips, @bitsawer, @hakro, @mihe, @nongvantinh

Thanks to all contributors and donors for making Godot possible!

akien-mga committed:
Merge pull request #78653 from dalexeev/unexpose-doc-cache

Unexpose `DocCache` class

akien-mga committed:
Merge pull request #78670 from KoBeWi/cache_with_full_power

Improve script icon cache

akien-mga committed:
Merge pull request #78709 from Faless/mp/4.x_fix_delta_again

[MP] Fix delta variables index decoding.

groud committed:
Explain a bit more RPCs options

RandomShaper committed:
Ensure default node groups' call queue are processed

akien-mga committed:
Merge pull request #78713 from RandomShaper/fix_default_group_cq

Ensure default node groups' call queue are processed

akien-mga committed:
Fixup compilation warning after #78713

akien-mga committed:
mbedtls: Improve X509 certificate load error handling

Positive numbers shouldn't be treated as a hard failure:

> For CRTs in PEM encoding, the function parses permissively:
> if at least one certificate can be parsed, the function
> returns the number of certificates for which parsing failed
> (hence \c 0 if all certificates were parsed successfully).
> If no certificate could be parsed, the function returns
> the first (negative) error encountered during parsing.

Fixes #77994.

KoBeWi committed:
Fix missing translation in SceneCreateDialog

RandomShaper committed:
Fix change of precedence in fixup

akien-mga committed:
Merge pull request #78719 from RandomShaper/fix_fixup

Fix change of precedence in fixup

akien-mga committed:
Merge pull request #78717 from KoBeWi/TracToR

Fix missing translation in SceneCreateDialog

akien-mga committed:
Bump version to 4.1-rc

Faster release cycles are proving to work fairly well, and we have a lot more
confidence label the current state as RC as we've had in the past.

Let's get to 4.1-stable quickly and reopen development!

akien-mga committed:
i18n: Sync translations with Weblate

clayjohn committed:
Use a filter with mipmaps when initializing textures with mipmaps in GL Compatibility renderer

This works around a driver bug in some older devices and should be harmless on any spec-compliant device

RedworkDE committed:
C#: Automatically generate version defines

YuriSizov committed:
Merge pull request #78720 from clayjohn/GLES3-tex-init

Use a filter with mipmaps when initializing textures with mipmaps in GL Compatibility renderer

rburing committed:
Fix CharacterBody3D get_position_delta()

Calculate the difference of global positions.
This also fixes get_real_velocity().

Rindbee committed:
Fix that `ViewportTexture` cannot be setup again after failed setup

`vp_pending` is a flag indicating to wait for `ViewportTexture::_setup_local_to_scene`
to be called. Once this method is called, no matter whether the setup will fail or not,
it needs to be reset to `false`.

Faless committed:
Merge pull request #78716 from akien-mga/mbedtls-load-error-handling

mbedtls: Improve X509 certificate load error handling

RandomShaper committed:
Fix node internal/non-internal processing order

Bonus:
Fix issues in group-processing related setters

akien-mga committed:
Merge pull request #78727 from rburing/really_velocity

Fix CharacterBody3D `get_position_delta()` and `get_real_velocity()`

CURR-E committed:
Correct description of ParticleProcessMaterial.lifetime_randomness

akien-mga committed:
Merge pull request #78728 from Rindbee/reset-vp_pending-on-failure-or-not

Fix that `ViewportTexture` cannot be setup again after failed setup

yvie-k committed:
Fix invalid minimum size for translated messages in option button

KoBeWi committed:
Fix missing translations in EventListenerLineEdit

akien-mga committed:
Merge pull request #78711 from groud/explain_rpc

Explain a bit more RPCs options

akien-mga committed:
Merge pull request #78745 from RandomShaper/fix_node_pr

Fix node processing order

akien-mga committed:
Merge pull request #78767 from KoBeWi/space_or_space

Fix missing translations in EventListenerLineEdit

aaronfranke committed:
Sort project tags before saving

NiskashY committed:
Fix disabled slider highlighting

timothyqiu committed:
Fix function signature in EditorImportPlugin example

RedworkDE committed:
C# Fix reloading of non-tool scripts

akien-mga committed:
Merge pull request #78787 from RedworkDE/net-non-tool-reload

C#: Fix reloading of non-tool scripts

akien-mga committed:
Merge pull request #78740 from CURR-E/lifetime_randomness

Correct description of `ParticleProcessMaterial.lifetime_randomness`

akien-mga committed:
Merge pull request #78784 from timothyqiu/editor-import-plugin-example

Fix function signature in `EditorImportPlugin` example

joemarshall committed:
fix threading bug in vulkan rendering device

akien-mga committed:
Project converter: Use same rendering driver as Project Manager

Which means by default OpenGL 3, but it can still be overridden from the command line.
Fixes #76303.

KoBeWi committed:
Fix crash with failed compatibility tiles

akien-mga committed:
Merge pull request #78796 from KoBeWi/vengeful_spirit_of_Godot3_tries_to_destroy_the_new_engine

Fix crash with failed compatibility tiles

Calinou committed:
Remove uses of `vformat()` with no placeholders

This is identical to passing the string directly.

KamilBrzoskowski committed:
Fix formatting of dlopen error message on Windows

And harmonize the format for all platforms.

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

CncealDVRflectN committed:
add explicit default initialization for flag in undo redo operation

akien-mga committed:
Fix missing GDExtension in-editor API reference

This commit partially reverts a change in e1ce0340b78875a864d449a5e3e38e4535e9a800
which would prevent from generating API reference for GDExtension APIs.

Fixes #78829.

akien-mga committed:
Revert "Update RVO2 to git 2022.09"

This reverts commit c92088110567bd9c61aa046e9a93bdbc6469073e.

Fixes #78826.

akien-mga committed:
Merge pull request #78830 from akien-mga/classref-fix-missing-gdextension-api

Fix missing GDExtension in-editor API reference

akien-mga committed:
Merge pull request #78831 from akien-mga/revert-78099

Revert "Update RVO2 to git 2022.09"

akien-mga committed:
i18n: Sync translations with Weblate

bruvzg committed:
[RTL] Fix multithreaded scrollbar visibility update.

akien-mga committed:
Merge pull request #78809 from CncealDVRflectN/fix-operation-flag-default-initialiation

Add explicit default initialization for flag in undo redo operation structure

akien-mga committed:
Merge pull request #78833 from bruvzg/scroll_fix

[RTL] Fix multithreaded scrollbar visibility update.

lewiji committed:
Return shader parse error when using 'hint_normal_roughness_texture' and not using the Forward+ backend

RandomShaper committed:
Fix zero-sized WorkerThreadPool not processing group tasks

Sauermann committed:
Fix `Node::add_sibling` parent check

Replace `data.blocked > 0` by `data.parent->data.blocked > 0` in order
to check if the parent is busy.

smix8 committed:
Fix NavigationAgent continues avoidance velocity

Fixes NavigationAgent continues avoidance velocity.

clayjohn committed:
Initialize particles instance buffer in case it is used before being updated

smix8 committed:
Fix NavigationAgent position not always updating

Fixes NavigationAgent position not always updating.

nklbdev committed:
Fix wrong type casting for octahedral tangents

Chaosus committed:
Fix using uint suffix at the hex number declaration in shaders

zorbathut committed:
Fix: Incorrect property names in FontFile::_get_property_list().

nklbdev committed:
Potencially fix nan's on octahedral tangents in RenderingServer

KoBeWi committed:
Fix dropping files from res:// to res://

Rindbee committed:
Make sure the shortcut key respects the context in `TileSetAtlasSourceEditor`

theraot committed:
Do not change a node unique name to the same name

neikeq committed:
C#: Fix NodePaths completion error for not calling from main thread

The node API we use for code completion changed and no longer allows
being called from threads other than the main one.

timothyqiu committed:
Fix error when non-ASCII characters in resource pack path

AThousandShips committed:
Add compatibility properties to `TouchScreenButton`

Added support for `3.x` properties:
* `normal` -> `texture_normal`
* `pressed` -> `texture_pressed`

timothyqiu committed:
Translate "No match" message in FindReplaceBar

Scony committed:
Fix crash in 'NavigationAgent3D', fixes #78910

anvilfolk committed:
Fix regression with enum descriptions now showing up in documentation.

bruvzg committed:
[RTL] Fix multithreaded scrollbar visibility update 2.

AThousandShips committed:
Clarify that `Gradient.colors/offsets` are returned by-copy

konczg committed:
Fix GDVIRTUAL_NATIVE_PTR

This commit fixes the usage of GDExtensionPtr and GDExtensionConstPtr
in GDExtension by introducing the required VariantInternalAccessor
specializations.

Sponsored by Migeran (https://migeran.com)

bitsawer committed:
Fix shader language float literal precision truncation

RandomShaper committed:
Fix management of safe-for-nodes flag in ResourceLoader and WorkerThreadPool

YuriSizov committed:
Prevent crash when processing line caches in RichTextLabel

RandomShaper committed:
Workaround resource loading crashes due to buggy TLS

KoBeWi committed:
Fix node selection not handled correctly at launch

akien-mga committed:
Merge pull request #78850 from smix8/navagent_velocity_fix_4.x

Fix NavigationAgent continues avoidance velocity

akien-mga committed:
Merge pull request #78857 from smix8/navagent_pos_fix_4.x

Fix NavigationAgent position not always updating

akien-mga committed:
Merge pull request #78939 from Scony/navi-crash-fix

Fix crash in `NavigationAgent3D`

akien-mga committed:
Merge pull request #78968 from bruvzg/scroll_fix2

[RTL] Fix multithreaded scrollbar visibility update 2.

akien-mga committed:
Merge pull request #78970 from AThousandShips/gradient_doc

Clarify that `Gradient.colors/offsets` are returned by-copy

akien-mga committed:
Merge pull request #78971 from kisg/fix_gdvirtual_native_ptr

Fix `GDVIRTUAL_NATIVE_PTR` by adding missing `VariantInternalAccessor` specializations

akien-mga committed:
Merge pull request #78974 from RandomShaper/fix_wtp_safe_flag

Fix management of safe-for-nodes flag in ResourceLoader and WorkerThreadPool

akien-mga committed:
Merge pull request #78975 from YuriSizov/rtl-fix-refactoring-typo

Prevent crash when processing line caches in `RichTextLabel`

akien-mga committed:
Merge pull request #78980 from KoBeWi/demise_of_a_fix

Fix node selection not handled correctly at launch

akien-mga committed:
Merge pull request #78928 from neikeq/issue-78913

C#: Fix NodePaths completion error for not calling from main thread

Amir-Rasteg committed:
Fix a typo in the `String.to_float` description

akien-mga committed:
Merge pull request #78977 from RandomShaper/workaround_tls_shutdown

Workaround resource loading crashes due to buggy TLS

smix8 committed:
Fix closest possible navigation path position

Fixes closest possible navigation path position.

bruvzg committed:
[macOS] Fix set_native_icon crash with empty or invalid ICNS file.

bruvzg committed:
[Windows] Fix setting initial non-exclusive window mode.

RedworkDE committed:
Fix export options of scripted EditorExportPlugins

sepTN committed:
fix typo on TLSOptions.xml and library_godot_display.js

capnm committed:
Fix import hints that are followed by dot.number.

Fixes #78881.

BastiaanOlij committed:
Take eye offset into account for depth in StandardMaterial3D

YuriSizov committed:
Add Godot 4.1 changelog

bruvzg committed:
[macOS/iOS] Set MoltenVK logging level based on `--verbose` flag.

akien-mga committed:
Merge pull request #79056 from YuriSizov/meta-add-4.1-changelog

Add Godot 4.1 changelog

YuriSizov committed:
Merge pull request #79050 from capnm/fix_teststr_what

Fix import hints that are followed by dot.number

akien-mga committed:
i18n: Sync translations with Weblate

akien-mga committed:
Bump version to 4.1-stable \o/

Around 1500 commits from 300+ contributors merged over 4 months.

The new 4.x release cycle with 3 months of development and 1 month of
bugfixing proved to work fairly well for this 4.1 release, and we will
keep refining it for future releases.

The faster-paced release cycle means that each minor 4.x release will
have a small scope and won't be as impressive as the massive 4.0 was,
but it means that users get access to the new features and bug fixes
faster, and the stabilization phase is also significantly shortened
(only one month of feature freeze, so contributors don't need to wait
long to see their approved feature PRs merged for the next milestone).

Onwards to 4.2!

akien-mga committed:
Bump version to 4.2-dev

Keep on waitin'

akien-mga committed:
Merge pull request #78722 from RedworkDE/net-auto-version-defines

C#: Automatically generate version defines

YuriSizov committed:
Bump version of module docs to 4.2

KoBeWi committed:
Collapse bottom panel if there is no active tab

Sauermann committed:
Revert "Fix focusloss of non-exclusive `AcceptDialog` with `close_on_escape`"

This reverts commit 7f547fcf09e7af0e2443356fe7a003c3c8335cd6.

Daylily-Zeleen committed:
Fixed the fallback logic of OS::shell_show_in_file_manager

mb4c committed:
Add tooltip description wrapping in scene tree and plugin settings

akien-mga committed:
Doctool: Remove version attribute from XML header

We don't use that info for anything, and it generates unnecessary diffs
every time we bump the minor version (and CI failures if we forget to
sync some files from opt-in modules (mono, text_server_fb).

raulsntos committed:
C#: Compare symbol names without null flow state

akien-mga committed:
Linux: Allow unbundling brotli to use system library

akien-mga committed:
Linux: Link libsquish directly when unbundling, .pc file unreliable

bitsawer committed:
Make shader preprocessor keyword colors consistent

aaronfranke committed:
Improve user-friendliness of project version mismatch message

aaronfranke committed:
Fix property hint class name type string restriction and replace mode

TokageItLab committed:
Fix infinity loop state can't break

akien-mga committed:
Merge pull request #78845 from RandomShaper/fix_wtp_zero_batch_lazy

Fix zero-sized WorkerThreadPool not processing group tasks

akien-mga committed:
Merge pull request #78852 from clayjohn/Particles-glitch

Initialize particles instance buffer in case it is used before being updated

akien-mga committed:
Merge pull request #78902 from nklbdev/Fix_wrong_type_casting_in_RenderingServer

Fix wrong type casting for octahedral tangents

akien-mga committed:
Merge pull request #78906 from Chaosus/shader_allow_hex_uint

Fix using uint suffix at the hex number declaration in shaders

akien-mga committed:
Merge pull request #78907 from zorbathut/pr_fontprop

Fix incorrect property names in `FontFile::_get_property_list()`

akien-mga committed:
Merge pull request #78914 from KoBeWi/lazy_drop

Fix dropping files from res:// to res://

akien-mga committed:
Merge pull request #78920 from Rindbee/fix-invalid-shortcut-keys-in-TileSetAtlasSourceEditor

Make sure the shortcut key respects the context in `TileSetAtlasSourceEditor`

akien-mga committed:
Merge pull request #78935 from timothyqiu/zip-pack

Fix error when non-ASCII characters in resource pack path

akien-mga committed:
Merge pull request #78938 from timothyqiu/no-match

Translate "No match" message in FindReplaceBar

akien-mga committed:
Merge pull request #78953 from anvilfolk/enum-desc

Fix regression with GDScript enum descriptions now showing up in documentation.

akien-mga committed:
Merge pull request #78972 from bitsawer/fix_shader_float_precision

Fix shader language float literal precision truncation

akien-mga committed:
Merge pull request #78989 from Amir-Rasteg/patch-1

Fix a typo in the `String.to_float` description

akien-mga committed:
Merge pull request #79010 from bruvzg/set_native_icon_crash

[macOS] Fix `set_native_icon` crash with empty or invalid ICNS file.

akien-mga committed:
Merge pull request #79025 from RedworkDE/scripted-export-plugin

Fix export options of scripted `EditorExportPlugin`s

akien-mga committed:
Merge pull request #79049 from BastiaanOlij/fix_stereo_depth_paralax

Take eye offset into account for depth in StandardMaterial3D

akien-mga committed:
Merge pull request #79084 from Sauermann/revert-78363

Revert "Fix focusloss of non-exclusive `AcceptDialog` with `close_on_escape`"

akien-mga committed:
Merge pull request #79087 from Daylily-Zeleen/daylily-zeleen/fix_shell_show_in_file_manager

Fix the fallback logic of `OS::shell_show_in_file_manager`.

akien-mga committed:
Merge pull request #79092 from akien-mga/doctool-remove-version-attribute

Doctool: Remove version attribute from XML header

akien-mga committed:
Merge pull request #79094 from raulsntos/dotnet/fix-symbol-comparison

C#: Compare symbol names without null flow state

akien-mga committed:
Merge pull request #79101 from akien-mga/linux-unbundle-brotli

Linux: Allow unbundling brotli to use system library

akien-mga committed:
Merge pull request #79112 from bitsawer/fix_preprocessor_keyword_colors

Make shader preprocessor keyword colors consistent

clayjohn committed:
Avoid freeze when interacting with menus on Wayland by re-aquiring next swapchain image after updating swapchain

RedworkDE committed:
C#: Add null check before calling `UnregisterGodotObject`

Owl-A committed:
add 3.x compatibility for animation loop mode

akien-mga committed:
Merge pull request #54071 from KoBeWi/infinite_unlimited_connections

Focus current node after connecting

akien-mga committed:
Merge pull request #75806 from bcmpinc/issue-61174

Fix Camera3D `project_*` methods not accounting for frustum offset

akien-mga committed:
Merge pull request #78078 from Sauermann/fix-drop-physics-mouse-over

Ensure that `_drop_physics_mouseover` only happens when necessary

akien-mga committed:
Merge pull request #78263 from bruvzg/fl_all

[Windows] Flash both the window caption and taskbar button on `request_attention`.

akien-mga committed:
Merge pull request #78522 from MewPurPur/improve-tileset-labels

Improve string drawing in the tiledata editor

akien-mga committed:
Merge pull request #78524 from MewPurPur/4920726f6c6c65642061203621

Fix tooltip of enum value without description

akien-mga committed:
Merge pull request #78775 from aaronfranke/proj-tag-sort

Sort project tags before saving

akien-mga committed:
Merge pull request #78925 from theraot/fix_rename_unique

Do not change a node unique name to the same name

akien-mga committed:
Merge pull request #79090 from mb4c/wrap-tooltip

Add tooltip description wrapping in scene tree and plugin settings

akien-mga committed:
Merge pull request #79141 from TokageItLab/fix-infinity-state-loop

Fix infinite loop state check in `AnimationStateMachine`

akien-mga committed:
Merge pull request #79151 from RedworkDE/net-unregister-godotobject-check

C#: Add null check before calling `UnregisterGodotObject`

jpcerrone committed:
Fix comments and indentation in .gdshaderinc files
Fixes #78205
The handling of comments and indentation in the shader editor
wasn't considering shader include files.

RedworkDE committed:
C#: Fix command line exporting

aaronfranke committed:
Fix incorrect documentation for `Engine.get_architecture_name()`

MewPurPur committed:
Fix erroneous pad_zeros warning

HolonProduction committed:
Update FileDialog button activity when file_mode is changed.

akien-mga committed:
Merge pull request #46125 from angad-k/fix-object-to-check-script-pointer-validity

Check parameter validity in `Object::set_script`

akien-mga committed:
Merge pull request #71024 from marzecdawid/better-cursor-behaviour-in-tree-with-ctrl

Fix cursor behavior for multiselect in Tree while holding CTRL

akien-mga committed:
Merge pull request #74012 from myaaaaaaaaa/remove-tap

Remove unused `threaded_array_processor.h`

akien-mga committed:
Merge pull request #78551 from AThousandShips/call_rpc

Fix `rpc` calls with binds

akien-mga committed:
Merge pull request #78624 from puchik/canvas-background-specular-ghosting

Clear specular buffer if sky mode is canvas and screen space effects are used

akien-mga committed:
Merge pull request #78631 from Calinou/shader-editor-creation-dialog-use-bullet-points

Use bullet points in shader editor creation dialog

akien-mga committed:
Merge pull request #78694 from Yadnesh-Kulkarni/bugfix-grid-view-disappear-#78588

Fix 3D viewport grid disappearing on scene tab changes

akien-mga committed:
Merge pull request #78701 from Calinou/joy-button-name-tweak-xbox-menu

Mention Xbox menu button by name in Start button description

akien-mga committed:
Merge pull request #78794 from joemarshall/patch-1

Fix threading bug in Vulkan rendering device

akien-mga committed:
Merge pull request #78795 from akien-mga/project-converter-keep-rendering-driver

Project converter: Use same rendering driver as Project Manager

akien-mga committed:
Merge pull request #78797 from Calinou/remove-redundant-vformat

Remove uses of `vformat()` with no placeholders

akien-mga committed:
Merge pull request #78832 from van800/path_locator_4x

Update the RiderPathLocator to support the JetBrains Toolbox 2.0

akien-mga committed:
Merge pull request #78835 from Dragoncraft89/patch-1

Fix invalid minimum size for translated messages in option button

akien-mga committed:
Merge pull request #78839 from lewiji/normal_roughness_mobile_fix

Fix invalid shader compilation when using `hint_normal_roughness_texture` in mobile backend

akien-mga committed:
Merge pull request #78847 from Sauermann/fix-sibling-fail

Fix `Node::add_sibling` parent check

akien-mga committed:
Merge pull request #78940 from AThousandShips/touch_compat

Add compatibility properties to `TouchScreenButton`

akien-mga committed:
Merge pull request #79016 from bruvzg/fix_win_initial_ne_fs

[Windows] Fix setting initial non-exclusive window mode.

akien-mga committed:
Merge pull request #79048 from sepTN/its-platform-not-plafrom

Fix a typo in the TLSOptions documentation

akien-mga committed:
Merge pull request #79061 from bruvzg/mvk_log_level

[macOS/iOS] Set MoltenVK logging level based on `--verbose` flag.

akien-mga committed:
Merge pull request #79105 from akien-mga/linux-libsquish-lacks-pkgconfig

Linux: Link libsquish directly when unbundling, .pc file unreliable

akien-mga committed:
Merge pull request #79158 from jpcerrone/shaderinclude_fixes

Fix comments and indentation in .gdshaderinc files

akien-mga committed:
Merge pull request #79173 from RedworkDE/net-commandline-exporting

C#: Fix command line exporting

akien-mga committed:
Merge pull request #79174 from aaronfranke/eng-get-arch-doc

Fix incorrect documentation for `Engine.get_architecture_name()`

akien-mga committed:
Merge pull request #79202 from MewPurPur/fix-pad-zeros-error

Fix erroneous `pad_zeros()` warning

sepTN committed:
Fix various typos in documentation

Rindbee committed:
Fix `PackedScene::get_last_modified_time()` always returns `0`

The variables operated by `PackedScene::set_last_modified_time()`
and `PackedScene::get_last_modified_time()` are different.

akien-mga committed:
Merge pull request #79143 from clayjohn/wayland-menu

Avoid freeze when interacting with menus on Wayland by re-aquiring next swapchain image after updating swapchain

akien-mga committed:
Merge pull request #79211 from HolonProduction/file-dialog-button-update

Update FileDialog button activity when file_mode is changed.

akien-mga committed:
Merge pull request #79223 from sepTN/some-typos-ifound-today

Fix various typos in documentation

cbscribe committed:
Fix rigid body property description

smix8 committed:
Add performance note for parsing source geometry

Adds performance note for parsing source geometry.

akien-mga committed:
Merge pull request #78802 from KamilBrzoskowski/fix-vformat-for-os_windows-file

Fix formatting of dlopen error message on Windows

akien-mga committed:
Merge pull request #79004 from smix8/fix_closest_navpath_pos_4.x

Fix closest possible navigation path position

akien-mga committed:
Merge pull request #79252 from smix8/parse_perf_warning_4.x

Add performance note for parsing source geometry

OverloadedOrama committed:
Fix `Camera2D.rotating` not being converted and reversed properly

Godot 3's Camera2D `rotating = true` and `rotating = false` are supposed to be converted and reversed to `ignore_rotation = false` and `ignore_rotation = true` respectively, but this wasn't the case before this PR, as the project converted was failing to properly read the `true` and `false` strings, thus resulting in `ignore_rotation = true` in all cases.

clayjohn committed:
Unify error condition for particles trail lifetime

akien-mga committed:
Merge pull request #79237 from Rindbee/fix-get_last_modified_time-returns-0-in-PackedScene

Fix `PackedScene::get_last_modified_time()` always returns `0`

akien-mga committed:
Merge pull request #79250 from cbscribe/rb_doc_fix

Fix rigid body `contact_monitor` property description

akien-mga committed:
Merge pull request #79270 from clayjohn/particle-trails-error

Unify error condition for particles trail lifetime

YuriSizov committed:
Merge pull request #78776 from NiskashY/fix-disabled-slider-highlighting

Fix disabled slider highlighting

akien-mga committed:
Change explicit 'Godot 4.0' references to 'Godot 4'

Fixes #79276.

akien-mga committed:
Merge pull request #79139 from aaronfranke/prop-hint-cl-na-type-str-restr-repl

Fix property hint class name type string restriction and replace mode

akien-mga committed:
Merge pull request #79155 from Owl-A/bugfix-57779

Add 3.x compatibility for animation loop mode

akien-mga committed:
Merge pull request #79264 from OverloadedOrama/3-to-4-camera2d-rotating

Fix `Camera2D.rotating` not being converted and reversed properly

akien-mga committed:
Merge pull request #79277 from akien-mga/godot-4-ever

Change explicit 'Godot 4.0' references to 'Godot 4'

dalexeev commented on modules/gdscript/gdscript_analyzer.cpp:
```diff
@@ -1255,6 +1255,19 @@ void GDScriptAnalyzer::resolve_function_body(GDScriptParser::FunctionNode *p_fun
+void GDScriptAnalyzer::resolve_getter_body(GDScriptParser::FunctionNode *p_getter) {
+	resolve_function_body(p_getter);
+
+	GDScriptParser::FunctionNode *previous_function = parser->current_function;
+	parser->current_function = p_getter;
```
I don't think a separate method is needed here. The current function context is not used below (lines 1264-1266). Also, in this case, you would also need to switch the static context (see #77151).

You should move lines 1264-1266 inside `resolve_class_body()` after calling `resolve_function_body()` on line 920 and remove the `resolve_getter_body()` method.

akien-mga committed:
Merge pull request #79118 from aaronfranke/friendly-upgrades

Improve user-friendliness of project version mismatch message

YuriSizov committed:
Merge pull request #79078 from KoBeWi/if_you_gaze_into_the_empty_bottom_panel,_the_empty_bottom_panel_gazes_also_into_you

Collapse bottom panel if there is no active tab

garychia committed:
Generate errors for getters that return no value

garychia committed:
Resolve conflicts

garychia commented:
I'll close this pull request and open a new one to work on the lastest codebase.
